### DISCUSSION TITLE
Dialog close when click outside

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#7069] [Help]
Hi,

is there a way to close dialogs when user click outside them?

![Schermata 2024-01-29 alle 15 40 13](https://github.com/serenity-is/Serenity/assets/14870650/2eadd725-ecb1-439e-a7f2-270318f26f69)

**Comment by VictorTomaili** [comment]
you can try to add click event for the backdrop in dialog constructors. Do not forget to `off` on dialog dispose. 
for example 
```js
$('.ui-widget-overlay').on('click', (e) => {
    console.log('ui-widget-overlay clicked');
})
```

**Reply by elenaaralla** [reply]
Hi @VictorTomaili,
what do you mean with "Do not forget to off on dialog dispose."

Thank you,
Elena.

**Reply by VictorTomaili** [reply]
when you register an event on dialog open, you need to remove this event while disposing the dialog (when closing). 
https://api.jquery.com/off/

```ts
protected destroy() {
//unregister the event 
}
```
